modules/survey_module/ajax/survey_api.php

```diff
@@ -0,0 +1,364 @@
+     * REQUEST.
+     *
+     * @return null
+     */
+    function initialize()
+    {
+        ob_start('ob_gzhandler');
+        $client = new NDB_Client();
+        $client->makeCommandLine();
+        $client->initialize();
+
+        $factory  = NDB_Factory::singleton();
+        $config   = $factory->config();
+        $settings = $factory->settings();
+
+        $this->tpl_data['baseurl'] = $settings->getBaseURL();
+        $this->key = $_REQUEST['key'];
+
+        $DB = Database::singleton();
+        $this->SurveyInfo = $DB->pselect(
+            "SELECT Status, Test_name, CommentID FROM participant_accounts
+            WHERE OneTimePassword=:key",
+            array('key' => $this->key)
+        );
+
+        if (empty($this->SurveyInfo)) {
+            throw new Exception("The given survey doesn't exist", 404);
+        } else if (count($this->SurveyInfo) > 1) {
+            throw new Exception("Well looks like we made a mistake :(", 500);
+        } else if ($this->SurveyInfo[0]['Status'] === 'Complete') {
+            throw new Exception("Data has already been submitted.", 403);
+        }
+
+        $this->SurveyInfo = $this->SurveyInfo[0];
+
+    }
+
+    /**
+     * Handles a request by delegating to the appropriate
+     * handle method
+     *
+     * @return null
+     */
+    function handleRequest()
+    {
+        switch($_SERVER['REQUEST_METHOD']) {
+        case 'GET':
+            $this->handleGET();
+            break;
+        case 'PUT':
+            $this->handlePUT();
+            break;
+        case 'PATCH':
+            $this->handlePATCH();
+            break;
+        case 'POST':
+            $this->handlePOST();
+            break;
+        default:
+            $this->header("HTTP/1.1 501 Not Implemented");
+            break;
+        }
+    }
+
+    /**
+     * Handle a GET request. This will render and display the page.
+     *
+     * @return null
+     */
+    function handleGET()
+    {
+
+        try {
+            $this->Instrument = \NDB_BVL_Instrument::factory(
+                $this->SurveyInfo['Test_name'],
+                $this->SurveyInfo['CommentID'],
+                null,
+                true
+            );
+        } catch(\Exception $e) {
+            throw new Exception("Instrument not found", 405);
+        }
+
+        $this->tpl_data['InstrumentJSON'] = $this->Instrument->toDirectJSON();
+
+        $Values = \NDB_BVL_Instrument::loadInstanceData($this->Instrument);
+
+        unset($Values['CommentID']);
+        unset($Values['UserID']);
+        // unset($Values['Testdate']);
+        unset($Values['Data_entry_completion_status']);
+
+        // Unset score values
+        $json_instrument = json_decode($this->tpl_data['InstrumentJSON']);
+        $this->unsetScores($Values, $json_instrument->Elements);
+
+        $this->tpl_data['Values'] = json_encode($Values);
+
+        echo json_encode($this->tpl_data);
```

Comment:
Since the entirety of `$this->tpl_data` is being json_encoded, is the json_encode on `129` necessary? I believe json_encode works recursively.